app/javascript/components/subject/RemoveFromHousehold.js

```diff
@@ -0,0 +1,123 @@
+    let current = this.state.showModal;
+    this.setState({
+      showModal: !current,
+    });
+  }
+
+  handleChange(event) {
+    this.setState({ [event.target.id]: event.target.value });
+  }
+
+  submit() {
+    this.setState({ loading: true }, () => {
+      axios.defaults.headers.common['X-CSRF-Token'] = this.props.authenticity_token;
+      axios
+        .post(window.BASE_PATH + '/patients/' + this.props.patient.id + '/update_hoh', {
+          new_hoh_id: this.props.patient.id,
+        })
+        .then(() => {
+          location.href = window.BASE_PATH + '/patients/' + this.props.patient.id;
+        })
+        .catch(error => {
+          console.error(error);
+        });
+    });
+  }
+
+  checkHouseholdRemoveEligible() {
+    let self = this;
+    axios
+      .get(window.BASE_PATH + '/patients/' + this.props.patient.id + '/household_removeable')
+      .then(response => {
+        self.setState({ removeEligible: response.data.removeable });
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }
+
+  createModal(title, toggle, submit) {
+    return (
+      <Modal size="lg" show centered>
+        <Modal.Header>
+          <Modal.Title>{title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form>
+            <Row>
+              <Form.Group as={Col}>
+                {this.state.removeEligible && (
+                  <Form.Label size="sm" className="nav-input-label">
+                    This monitoree will be removed from their household and will be responsible for their own responses.
+                  </Form.Label>
+                )}
+                {this.state.removeEligible == false && (
+                  <Form.Label size="sm" className="nav-input-label">
+                    This monitoree cannot be removed from their household until their email and primary telephone number differ from those of the current head
+                    of household.
+                  </Form.Label>
+                )}
+              </Form.Group>
+            </Row>
+          </Form>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="primary btn-square" onClick={submit} disabled={!this.state.removeEligible || this.state.loading}>
+            {this.state.loading && (
+              <React.Fragment>
+                <span className="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>&nbsp;
+              </React.Fragment>
+            )}
+            Remove
+          </Button>
+          <Button variant="secondary btn-square" onClick={toggle}>
+            Cancel
+          </Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  }
+
+  render() {
+    return (
+      <React.Fragment>
+        <Button size="sm" className="my-2" onClick={this.toggleModal}>
+          <i className="fas fa-house-user"></i> Remove From Household
+        </Button>
+        {this.state.removeEligible == undefined && this.checkHouseholdRemoveEligible()}
```

Comment:
`this.checkHouseholdRemoveEligible` is asynchronous, I'd like to try to avoid using it like this. Could we alternatively add the backend `household_removeable` logic as a patient method instead, and have its value passed to the show view, where it can be used immediately in this render logic?

Reply:
I considered that, my thought was that this functionality will only be for dependents so it's not worth doing the check and sending the data along for every single patient sent to the frontend. I figured that since remove_from_household is not a core-function that will be used all the time then it's probably okay if someone needed extra time to be available.

Reply:
Okay, then you need to make this call in `componentDidMount` https://reactjs.org/docs/react-component.html#componentdidmount, not inside render.